src/components/forms/checkbox/Checkbox.vue

```diff
@@ -0,0 +1,233 @@
+<script lang="ts" setup>
+import { type ContextColorsType } from '@/consts/colors';
+import Icon from '@/components/icons/Icon.vue';
+
+const props = withDefaults(
+  defineProps<{
+    value?: boolean;
+    indeterminate?: boolean;
+    disabled?: boolean;
+    color?: ContextColorsType;
+    sm?: boolean;
+    lg?: boolean;
+    hint?: string;
+    errorMessages?: string[];
+    hideDetails?: boolean;
+  }>(),
+  {
+    value: false,
+    indeterminate: false,
+    disabled: false,
+    color: 'primary',
+    sm: false,
+    lg: false,
+    hint: '',
+    errorMessages: () => [],
+    hideDetails: false,
+  }
+);
+
+const { sm, lg } = toRefs(props);
+
+const emit = defineEmits<{
+  (e: 'input', value: boolean): void;
+}>();
+
+const input = (event: Event) => {
+  const checked = (event.target as HTMLInputElement).checked;
+  emit('input', checked);
+};
+
+const iconSize: ComputedRef<number> = computed(() => {
+  if (get(lg)) {
+    return 28;
+  }
+  if (get(sm)) {
+    return 20;
+  }
+  return 24;
+});
+
+const css = useCssModule();
+const attrs = useAttrs();
```

Comment:
Is there a way in vue 3 to get only the class out for attrs (I remember reading something related)?
If so wouldn't it be better if classes were applied to the root div